File: JavaScript/6js 2/matrixAddition.js
/*
|--------------------------------------------------------------------------
| Matrix Addition
|--------------------------------------------------------------------------
|
*/

function matrixAddition(a, b) {
   let res = [];
   for (let i = 0; i < a.length; i++) {
      let row = [];
      for (let j = 0; j < a.length; j++) {
         row.push(a[i][j] + b[i][j]);
      }
      res.push(row);
   }
   return res;
}


const R = require("ramda"),
   matrixAddition = R.zipWith(R.zipWith(R.add));


function matrixAddition(a, b) {
   return a.map((a, i) => a.map((a, j) => a + b[i][j]));
}


function matrixAddition(a, b) {
   return a.map(function (row, i) {
      return row.map(function (col, ii) {
         return col + b[i][ii];
      });
   });
}

File: Python/6py 2/matrix_addition.py
"""
|--------------------------------------------------------------------------
| Matrix Addition
|--------------------------------------------------------------------------
|
"""

def matrix_addition(a, b):
   res = []
   for i in range(len(a)):
      row = []
      for j in range(len(a[i])):
         row.append(a[i][j] + b[i][j])
      res.append(row)
   return res


def matrix_addition(a, b):
   return [[sum(xs) for xs in zip(ra, rb)] for ra, rb in zip(a, b)]


from numpy import matrix
def matrix_addition(a, b):
   return (matrix(a) + matrix(b)).tolist()